source/elements/oneDAL/source/algorithms/clustering/kmeans.rst

```diff
@@ -1,153 +1,494 @@
+by the user.
+
+-------------
+Usage example
+-------------
+::
+
+   onedal::kmeans::model run_training(const onedal::table& data,
+                                      const onedal::table& initial_centroids) {
+
+      const auto kmeans_desc = onedal::kmeans::desc<float>{}
+         .set_cluster_count(10)
+         .set_max_iteration_count(50)
+         .set_accuracy_threshold(1e-4);
+
+      const auto result = onedal::train(kmeans_desc, data, initial_centroids);
+
+      print_table("labels", result.get_labels());
+      print_table("centroids", result.get_model().get_centroids());
+      print_value("objective", result.get_objective_function_value());
+
+      return result.get_model();
+   }
+
+
+::
+
+   onedal::table run_inference(const onedal::kmeans::model& model,
+                               const onedal::table& new_data) {
+
+      const auto kmeans_desc = onedal::kmeans::desc<float>{}
+         .set_cluster_count(model.get_cluster_count());
+
+      const auto result = onedal::infer(kmeans_desc, model, new_data);
+
+      print_table("labels", result.get_labels());
+   }
 
-where :math:`d(x_i, C)` is the distance from :math:`x_i` to the closest center
-in :math:`C`.
 
 ---
 API
 ---
 
-Descriptor
-----------
-.. code-block:: c++
-
-   namespace onedal::kmeans {
+Methods
+-------
+::
 
    namespace method {
-   struct lloyd {};
-   using by_default = lloyd;
+      struct lloyd {};
+      using by_default = lloyd;
    } // namespace method
 
+.. namespace:: onedal::kmeans::method
+.. struct:: lloyd
+
+   Tag-type that denotes `Lloyd's method`_.
+
+
+.. type:: by_default = lloyd
+
+   Alias tag-type for the Lloyd's method.
+
+
+Descriptor
+----------
+::
+
    template <typename Float = float,
              typename Method = method::by_default>
    class desc {
    public:
-      double get_gamma() const;
+      desc();
+
+      int64_t get_cluster_count() const;
+      int64_t get_max_iteration_count() const;
       double get_accuracy_threshold() const;
-      std::int64_t get_cluster_count() const;
-      std::int64_t get_max_iteration_count() const;
 
-      desc& set_gamma(double);
+      desc& set_cluster_count(int64_t);
+      desc& set_max_iteration_count(int64_t);
       desc& set_accuracy_threshold(double);
-      desc& set_cluster_count(std::int64_t);
-      desc& set_max_iteration_count(std::int64_t);
    };
 
-   } // namespace onedal::kmeans
+.. namespace:: onedal::kmeans
+.. class:: template<typename Float = float, \
+                    typename Method = method::by_default> \
+           desc
+
+   :tparam Float: The floating-point type that the algorithm uses for
+                  intermediate computations. Can be :expr:`float` or :expr:`double`.
+
+   :tparam Method: Tag-type that specifies variant of K-Means algorithm. Can be
+                   :expr:`method::lloyd` or :expr:`method::by_default`.
+
+   .. function:: desc()
+
+      Creates new instance of descriptor with the default attribute values.
```

Comment:
'Create**s** is aligned with TBB style of description, VPL seems to do 'create'
TBB uses 'constructs'. Looks like we will need simple guidelines or a decision each library can do it differently

Reply:
a related q of style: in the output there is no visible difference between functions and members, but a description of a function begins with the verb (which makes sense), while the members describe entities (also makes sense). This confuses me as a reader. @rscohn2  I'd suggest updating css to differentiate functions and members.